pkg/ingester/ingester.go

```diff
@@ -569,6 +569,27 @@ func (i *Ingester) GetOrCreateInstance(instanceID string) *instance { //nolint:r
 	return inst
 }
 
+func (i *Ingester) Ack(ctx context.Context, req *logproto.AckRequest) (*logproto.AckResponse, error) {
+	instanceID, err := tenant.TenantID(ctx)
+	if err != nil {
+		return &logproto.AckResponse{}, err
+	}
+
+	instance := i.GetOrCreateInstance(instanceID)
+	if err != nil {
+		return &logproto.AckResponse{}, err
+	}
+
+	instance.queryMtx.Lock()
+	queryIngester := instance.queries[req.Id]
+	instance.queryMtx.Unlock()
+	if queryIngester != nil {
+		queryIngester.ReleaseAck()
```

Comment:
hmm nitpicking maybe you should call it `SendAck()` instead of `ReleaseAck()`? (since SendAck will send an empty struct message to the channel)